command/connect/envoy/envoy.go

```diff
@@ -406,58 +393,14 @@ func (c *cmd) templateArgs() (*BootstrapTplArgs, error) {
 	httpCfg := api.DefaultConfig()
 	c.http.MergeOntoConfig(httpCfg)
 
-	// Trigger the Client init to do any last-minute updates to the Config.
+	// api.NewClient normalizes some values (Token, Scheme) on the Config.
 	if _, err := api.NewClient(httpCfg); err != nil {
```

Comment:
This call to `api.NewClient` still feels very brittle to me. I suspect some portion of the logic in `NewClient` either belongs in `DefaultConfig()`, or as methods on `Config`. I deferred making larger changes here for now, but I think it will be necessary soon.